apps/src/javalab/JavabuilderConnection.js

```diff
@@ -31,6 +31,13 @@ export default class JavabuilderConnection {
   // Get the access token to connect to javabuilder and then open the websocket connection.
   // The token prevents access to our javabuilder AWS execution environment by un-verified users.
   connectJavabuilder() {
+    // Don't attempt to connect to Javabuilder if we do not have a project identifier.
+    // This typically occurs if the project has not been modified from the starter code.
+    if (project.getCurrentId() === undefined) {
```

Comment:
should this change be here? Also we do have many cases where we want to run code that hasn't been edited (for example, run the starter code to see what happens)

Reply:
Yeah, I just decided to do these both in one to avoid merge conflicts. Please take another look!

I should rephrase this comment, the most common error case where this happens is a teacher viewing a student's work where the student has not edited the code and we have not set the channelId. This does not happen for students trying to run starter code...I figured out why this was the case at some point, let me see if I can dig that up...

Reply:
For future reference, I think we do not set a channelId when a teacher is viewing a student's (unmodified) code because we use this method to set the channelId:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/helpers/levels_helper.rb#L182

In the case of a teacher viewing a student's work, the @user variable represents the student to be viewed. When a student is viewing their own work, @user is nil, which results in the else condition being executed here, which creates a new channel token for them:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/helpers/levels_helper.rb#L88

Reply:
cool, I think updating the comment to explain this is only for read only mode would be good

Reply:
@molly-moen updated!